tests/unit/test_time_parser.py

```diff
@@ -14,187 +14,197 @@
+             ("R5//P1D", "R5/19991226T0930Z/P1D",
+                 ["19991226T0930Z", "19991227T0930Z",
+                  "19991228T0930Z", "19991229T0930Z",
+                  "19991230T0930Z"]),
+             ("R1", "R1/19991226T0930Z/P0Y",
+                 ["19991226T0930Z"])]
+    for test in tests:
+        if len(test) == 2:
+            expression, ctrl_data = test
+            ctrl_results = None
+        else:
+            expression, ctrl_data, ctrl_results = test
+        recurrence = (parsers[0].parse_recurrence(expression))[0]
+        test_data = str(recurrence)
+        assert test_data == ctrl_data
+        if ctrl_results is None:
+            continue
+        test_results = []
+        for i, test_result in enumerate(recurrence):
+            if i > len(ctrl_results) - 1:
+                break
+            assert str(test_result) == ctrl_results[i]
+            test_results.append(str(test_result))
+        assert test_results == ctrl_results
+
 
-    def test_first_recurrence_format(self):
-        """Test the first ISO 8601 recurrence format."""
-        tests = [("R5/T06/04T1230-0300",
-                  0,
-                  "R5/19991227T0600Z/20010604T1530Z",
-                  ["19991227T0600Z", "20000506T1422Z",
-                   "20000914T2245Z", "20010124T0707Z",
-                   "20010604T1530Z"]),
-                 ("R2/-0450000101T04Z/+0020630405T2200-05",
-                  2,
-                  "R2/-0450000101T0400Z/+0020630406T0300Z",
-                  ["-0450000101T0400Z", "+0020630406T0300Z"]),
-                 ("R10/T12+P10D/-P1Y",
-                  0,
-                  "R10/20000105T1200Z/20000506T1000Z",
-                  ["20000105T1200Z", "20000119T0106Z",
-                   "20000201T1413Z", "20000215T0320Z",
-                   "20000228T1626Z", "20000313T0533Z",
-                   "20000326T1840Z", "20000409T0746Z",
-                   "20000422T2053Z", "20000506T1000Z"])]
-        for test in tests:
-            if len(test) == 3:
-                expression, num_expanded_year_digits, ctrl_data = test
-                ctrl_results = None
-            else:
-                expression, num_expanded_year_digits, ctrl_data = test[:3]
-                ctrl_results = test[3]
-            parser = self._parsers[num_expanded_year_digits]
-            recurrence = parser.parse_recurrence(expression)[0]
-            test_data = str(recurrence)
-            self.assertEqual(test_data, ctrl_data)
-            if ctrl_results is None:
-                continue
-            test_results = []
-            for i, test_result in enumerate(recurrence):
-                self.assertEqual(str(test_result), ctrl_results[i])
-                test_results.append(str(test_result))
-            self.assertEqual(test_results, ctrl_results)
+def test_fourth_recurrence_format(parsers):
+    """Test the fourth ISO 8601 recurrence format."""
+    tests = [("PT6H/20000101T0500Z", "R25/19991226T0500Z/PT6H"),
```

Comment:
Can we not turn this into pytest parameters? 
If one fails it will make it easier to see _which_ one.

Reply:
Effort

Reply:
(fair justification IMO)

Reply:
No worries.